File: CSharp/13. Roman to Integer.cs
// 13. Roman to Integer
// Level: Easy
// https://leetcode.com/problems/roman-to-integer/

public class Solution
{
  public int RomanToInt(string s)
  {
    int total = 0;
    var map = new Dictionary<char, int>
    {
      { 'I', 1 },
      { 'V', 5 },
      { 'X', 10 },
      { 'L', 50 },
      { 'C', 100 },
      { 'D', 500 },
      { 'M', 1000 }
    };
    for (int i = 0; i < s.Length; i++)
    {
      if (i < s.Length - 1)
      {
        if (map[s[i]] < map[s[i + 1]])
          total -= map[s[i]];
        else
          total += map[s[i]];
      }
      else
        total += map[s[i]];
    }

    return total;
  }
}

File: Python/13. Roman to Integer.py
# 13. Roman to Integer
# Level: Easy
# https://leetcode.com/problems/roman-to-integer/

class Solution:
  def romanToInt(self, s: str) -> int:
    total = 0
    map = {'I': 1,
           'V': 5,
           'X':10,
           'L':50,
           'C':100,
           'D':500,
           'M':1000
           }
    for idx, x in enumerate(s):
      if idx < len(s) - 1:
        if map[x] < map[s[idx+1]]:
          total -= map[x] 
        else:
          total += map[x]
      else:
        total += map[x]

    return total
